src/main/java/com/dongsoop/dongsoop/chat/validator/ChatValidator.java

```diff
@@ -113,27 +173,27 @@ private Set<String> extractMessageIds(List<ChatMessage> messages) {
                 .collect(Collectors.toSet());
     }
 
-    private List<ChatMessage> filterMessages(List<ChatMessage> messages, Predicate<ChatMessage> filter) {
-        return messages.stream()
-                .filter(filter)
+    private List<ChatMessage> filterNewMessages(List<ChatMessage> clientMessages, Set<String> existingIds) {
+        Predicate<ChatMessage> isNewMessage = msg -> !existingIds.contains(msg.getMessageId());
+
+        return clientMessages.stream()
+                .filter(isNewMessage)
                 .toList();
     }
 
-    public void validateManagerPermission(ChatRoom room, Long userId) {
-        Long managerId = room.getManagerId();
-        if (managerId != null && !managerId.equals(userId)) {
-            throw new UnauthorizedManagerActionException();
-        }
+    private void validateUserExistsInRoom(ChatRoom room, Long userToKick) {
+        Optional.of(!room.getParticipants().contains(userToKick))
+                .filter(Boolean::booleanValue)
+                .ifPresent(ignored -> {
+                    throw new UserNotInRoomException();
+                });
     }
 
-    public void validateKickableUser(ChatRoom room, Long userToKick) {
-        if (!room.getParticipants().contains(userToKick)) {
-            throw new UserNotInRoomException();
-        }
-
-        Long managerId = room.getManagerId();
-        if (managerId != null && managerId.equals(userToKick)) {
-            throw new ManagerKickAttemptException();
-        }
+    private void validateNotKickingManager(ChatRoom room, Long userToKick) {
+        Optional.ofNullable(room.getManagerId())
+                .filter(managerId -> managerId.equals(userToKick))
+                .ifPresent(ignored -> {
+                    throw new ManagerKickAttemptException();
+                });
```

Comment:
```suggestion
Long managerId = room.getManagerId();
if (managerId.equals(userToKick)) {
  throw new ManagerKickAttemptException();
}
```

불필요한 클래스 래핑을 줄이고, 직관적으로 코드를 작성하는 것이 좋아보입니다.